### PR TITLE
Bug 2059674: Adjusts the Dynamic Demo Plugin and the SDK to meet on the usecases

### PR DESCRIPTION
**Pull request by andrewballantyne**
Fixes build issues with the Dynamic Demo Plugin.

`K8sAPIConsumer.tsx` Formatting changes - Not sure why, but our demo plugin wasn't lint'ed/prettier'ed to the same structure as the monorepo. Seems like an oversight but probably because of configuration issues. My IDE caught this as it's configured to look for it and apply it when applicable. I can undo the formatting changes if needed.

Odd our Delete & List commands had required properties. Adjusted to fix them (the real issue).

### CODE REVIEW

File: dynamic-demo-plugin/src/components/k8sConsumer/k8s-data.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const mockDeploymetData = {
             image: 'image-registry.openshift-image-registry.svc:5000/openshift/httpd:latest',
             ports: [
               {
+                protocol: 'tcp',
```
Comment (andrewballantyne) [inline dynamic-demo-plugin/src/components/k8sConsumer/k8s-data.ts:31]: This is a required type for DeploymentKind... I found it while mucking around. But we don't want to import the type as it causes the break between monorepo and this test plugin.

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts [hunk 2]
```diff
@@ -320,8 +320,8 @@ export const k8sKill = <R extends K8sResourceCommon>(
 type OptionsDelete<R> = BaseOptions & {
   model: K8sModel;
   resource: R;
-  requestInit: RequestInit;
-  json: Record<string, any>;
+  requestInit?: RequestInit;
+  json?: Record<string, any>;
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:324]: Not required by the underly method... optional should be here.

File: frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts [hunk 3]
```diff
@@ -391,7 +391,7 @@ export const k8sList = (
 type OptionsList = {
   model: K8sModel;
   queryParams: { [key: string]: any };
-  requestInit: RequestInit;
+  requestInit?: RequestInit;
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts:394]: Same here.

File: dynamic-demo-plugin/src/components/k8sConsumer/K8sAPIConsumer.tsx [hunk 4]
```diff
@@ -25,148 +35,161 @@ const K8sAPIConsumer: React.FC = () => {
...
-    );
-  }
+    k8sList({ model: k8sModel, queryParams: { ns: 'default' } })
+      .then((response) => {
+        setErrData('');
+        if (Array.isArray(response)) {
+          setK8sListData(response);
+        } else {
+          setK8sListData(response.items);
+        }
```
Comment (andrewballantyne) [inline dynamic-demo-plugin/src/components/k8sConsumer/K8sAPIConsumer.tsx:114]: Had to make this change as the TS compiler didn't like me assuming the type.

**Review by andrewballantyne — COMMENTED** [review]
Review breakdown.

File: dynamic-demo-plugin/src/components/k8sConsumer/K8sAPIConsumer.tsx [hunk 5]
```diff
@@ -25,148 +35,161 @@ const K8sAPIConsumer: React.FC = () => {
...
-    .catch((e) => {
-      setErrData(e.message)
-      console.error(e);
-     }
-    );
-  }
+    k8sList({ model: k8sModel, queryParams: { ns: 'default' } })
+      .then((response) => {
+        setErrData('');
+        if (Array.isArray(response)) {
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply dynamic-demo-plugin/src/components/k8sConsumer/K8sAPIConsumer.tsx:110]: @spadgett I think what we want is what we will be delivering [from the Core SDK](https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-utils/src/k8s/k8s-resource.ts#L130-L154) -- I could make two methods, but it seems TypeScript only infers so much.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by andrewballantyne** [comment]
Well that's an ugly looking failure:

```
>> An error was thrown in an afterAll
   StaleElementReferenceError: stale element reference: element is not attached to the page document
     (Session info: headless chrome=94.0.4606.81)
     (Driver info: chromedriver=94.0.4606.61 (418b78f5838ed0b1c69bb4e51ea02[521](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console/11116/pull-ci-openshift-console-master-e2e-gcp-console/1499751040054988800#1:build-log.txt%3A521)71854915-refs/branch-heads/4606@{#1204}),platform=Linux 4.18.0-305.34.2.el8_4.x86_64 x86_64)
       at Object.checkLegacyResponse /go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/error.js:546:15
       at parseHttpResponse /go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/http.js:509:13
       at /go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/http.js:441:30
       at runMicrotasks <anonymous>:null:null
       at processTicksAndRejections internal/process/task_queues.js:93:5
```
We should have proper page object scope coverage in our tests.

But I do not believe this is related to my work.
/retest

**Pull request merged**